graph/telemetry/istio/appender/util.go

```diff
@@ -11,20 +11,21 @@ import (
 	"github.com/kiali/kiali/graph"
 	"github.com/kiali/kiali/log"
 	"github.com/kiali/kiali/prometheus/internalmetrics"
+	"strings"
 )
 
 // package-private util functions (used by multiple files)
 
-func promQuery(query string, queryTime time.Time, api prom_v1.API, a graph.Appender) model.Vector {
-	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
This cancel has not make too much sense as context.Background() is empty and it's not cancellable or used to sync with other "context" area.

Reply:
Using WithCancel() to wrap context.Background() just provides the ability to cancel.  It's there *because* context.Background() is empty.   I still think we want to be able to cancel prom queries and when I wrote this I probably expected we would look into that sort of thing.  I agree that at the moment it's likely not needed, as we have no support for actually canceling the query, and as such the deferred cancel() is always invoked.  So for now I guess we can remove it, although the cost is nominal.